e2e/rnTestUtilities.js

```diff
@@ -33,49 +36,32 @@ const assertAndroidEvent = async (event, check) => {
   assertEvent(err, res, check);
 };
 
-const assertAndroidAutotrackHierarchy = async (
-  expectedName,
-  expectedHierarchy,
-  expectedTargetText
-) => {
+const assertAndroidAutotrackHierarchy = async (expectedName, expectedProps) => {
   return assertAndroidEvent({
     envId: HEAP_ENV_ID,
     event: {
       custom: {
         name: expectedName,
-        properties: {
-          touchableHierarchy: {
-            string: expectedHierarchy,
-          },
-          targetText: {
-            string: expectedTargetText,
-          },
-        },
+        properties: _.mapValues(expectedProps, value => {
+          string: value;
+        }),
       },
     },
   });
 };
 
-const assertAutotrackHierarchy = async (
-  expectedName,
-  expectedHierarchy,
-  expectedTargetText
-) => {
+const assertAutotrackHierarchy = async (expectedName, expectedProps) => {
   if (device.getPlatform() === 'android') {
-    return assertAndroidAutotrackHierarchy(
-      expectedName,
-      expectedHierarchy,
-      expectedTargetText
-    );
+    return assertAndroidAutotrackHierarchy(expectedName, expectedProps);
   } else if (device.getPlatform() === 'ios') {
     return assertIosPixel({
       t: expectedName,
-      k: [
-        'touchableHierarchy',
-        expectedHierarchy,
-        'targetText',
-        expectedTargetText,
-      ],
+      // Convert { key1: 'value1', key2: 'value2'} to ['key1', 'value1', 'key2', 'value2'] for
+      // custom props.
+      k: _(expectedProps)
+        .map((value, key) => [key, value])
+        .flatten()
```

Comment:
Could this be `_.flatMap(expectedProps, (v,k) => [k,v])` instead?

Reply:
TIL about `_.flatMap`! 👌